src/sif/bicyclecost.cc

```diff
@@ -566,18 +566,22 @@ Cost BicycleCost::EdgeCost(const baldr::DirectedEdge* edge,
                            const baldr::GraphTile* tile,
                            const uint32_t seconds) const {
   auto speed = tile->GetSpeed(edge, flow_mask_, seconds);
+  assert(speed < speedfactor_.size());
+  float sec = (edge->length() * speedfactor_[speed]);
```

Comment:
unfortunately these changes seem to be a bit problematic in that they return the wrong number of seconds along the edge if its one of the cases below (steps or ferry). seems we need to maybe just put a ternary in each return for the cost

Reply:
thx, I'll have a look tmrw morning at this logic again

Reply:
Oh hell yeah, there was actually a lot wrong in ped/bike costings.. Must've been a serious case of copy/paste dumbness. I went over all costings again and fixed mostly ped/bike